injected/src/features/broker-protection/actions/expectation.js

```diff
@@ -1,23 +1,43 @@
 import { getElement } from '../utils.js'
 import { ErrorResponse, SuccessResponse } from '../types.js'
+import { execute } from '../execute.js'
 
 /**
  * @param {Record<string, any>} action
- * @param {Document | HTMLElement} root
+ * @param {Record<string, any>} userData
+ * @param {Document} root
  * @return {import('../types.js').ActionResponse}
  */
-export function expectation (action, root = document) {
+export function expectation (action, userData, root = document) {
+    /**
+     * @type {Array<import('../types.js').BooleanResult & { failSilently?: boolean }>}
+     */
     const results = expectMany(action.expectations, root)
+    const errors = []
+    let runActions = true
+
+    results.forEach((x) => {
+        if (x.result === false) {
+            runActions = false
```

Comment:
this ends up being 'if any result is false, dont run' - that can be represented as this instead - to prevent needing these inner/outer values.

```js
const runActions = results.every(x => x.result === true);
```